src/nplinker/strain_collection.py

```diff
@@ -95,19 +95,21 @@ def lookup_index(self, index: int) -> Strain:
         """
         return self._lookup_indices[index]
 
-    def lookup(self, strain_id: str, default=None) -> Strain|None:
+    def lookup(self, strain_id: str) -> Strain:
         """Check whether the strain id is contained in the lookup table. If so, return the strain, otherwise return `default`.
 
+        Raises:
+            Exception if strain_id is not found.
+
         Args:
             strain_id(str): Strain id to lookup.
-            default (Strain, optional): Object to return otherwise. Defaults to None.
 
         Returns:
-            Optional[Strain]: Strain retrieved during lookup or object passed as default.
+            Strain: Strain retrieved during lookup or object passed as default.
         """
         if strain_id not in self._lookup:
             # logger.error('Strain lookup failed for "{}"'.format(strain_id))
-            return default
+            raise Exception(f'Strain lookup failed for "{strain_id}"')
```

Comment:
Narrow the `Exception` to `KeyError`? the `self._lookup` is a dict

Reply:
See https://github.com/NPLinker/nplinker/pull/91/commits/190c433171355a1bd7607caba32397135292d7eb